Extension/src/LanguageServer/configurations.ts

```diff
@@ -317,6 +321,11 @@ export class CppProperties {
         } else {
             configuration.includePath = [defaultFolder];
         }
+
+        if (settings.addNodeAddonIncludePaths) {
+            configuration.includePath = configuration.includePath.concat(this.nodeAddonIncludes);
```

Comment:
i can remove but if i do it creates this problem (which maybe i should be handling in some other way):

- start with "Add Node Addon Include Paths" checked - no squiggles
- uncheck "Add Node Addon Include Paths" - get squiggles
- check "Add Node Addon Include Paths" - squiggles do not go away

with the lines as they are the squiggles go away when the setting is checked.

Reply:
just need a simple project depending on `node-addon-api` with an `include <napi.h>` in a c/c++ file.

Reply:
could change it to `configuration.includePath.concat(this.nodeAddonIncludes.filter(i => configuration.includePath.indexOf(i) < 0)` to avoid duplicating if there's not a better way.

Reply:
i will figure it out. it's not clear why that makes a difference but it's not working the way i thought it was.

Reply:
i believe it's fixed now. the way it was working was accidental - not what i thought. thanks for the note.

not sure i handled the possibly undefined `configuration.includePath` in a standard way. let me know if you'd like it changed. i appreciate your time on this - i have learned a lot stumbling around with your help.